src/Google.Cloud.Functions.Hosting/HostingInternals.cs

```diff
@@ -171,6 +172,19 @@ internal static IServiceCollection AddServicesForFunctionTarget(IServiceCollecti
                     .AddSingleton<CloudEventFormatter>(new JsonEventFormatter())
                     .AddScoped(typeof(ICloudEventFunction), functionType);
             }
+            else if (GetGenericInterfaceImplementationTypeArguments(functionType, typeof(ITypedFunction<,>)) is Type[] signatureTypes)
+            {
+                Type argType = signatureTypes[0];
+                Type retType = signatureTypes[1];
+
+                services
+                    .AddScoped(typeof(IHttpFunction), typeof(TypedFunctionAdapter<,>).MakeGenericType(signatureTypes))
+                    .AddScoped(typeof(ITypedFunction<,>).MakeGenericType(signatureTypes), functionType);
+
+                Type jsonFormatImpl = typeof(JsonFormatImpl<,>).MakeGenericType(argType, retType);
```

Comment:
I've added a comment here to indicate that users can specify their own readers/writers, and added a new test for that.